packages/core/entity/AnnotationFormatter/duration-formatter.ts

```diff
@@ -19,12 +20,22 @@ export default {
         addUnit(msInAnHour, "H");
         addUnit(msInAMinute, "M");
         // Since seconds can have values like "3.5S", don't use floor when calculating
-        addUnit(1000, "S", false);
+        addUnit(msInASecond, "S", false);
 
         return display.trim();
     },
 
     valueOf(value: any) {
+        const RANGE_OPERATOR_REGEX = /([0-9]+)D ([0-9]+)H ([0-9]+)M ([0-9]*\.?[0-9]+)S/g;
+        const exec = RANGE_OPERATOR_REGEX.exec(value);
```

Comment:
Maybe something like `matches` is  more clear than `exec`?

Reply:
fixed in [761fa46](https://github.com/AllenInstitute/biofile-finder/pull/354/commits/761fa46a0c786142ff324d5007c58dc293e24b0b)